Tests/MeiliSearchIntegrationTests/KeysTests.swift

```diff
@@ -5,69 +5,93 @@ import Foundation
 // swiftlint:disable force_try
 class KeysTests: XCTestCase {
   private var client: MeiliSearch!
-  private var key: String = ""
   private var session: URLSessionProtocol!
 
   // MARK: Setup
 
   override func setUp() {
     super.setUp()
+
     session = URLSession(configuration: .ephemeral)
     client = try! MeiliSearch(host: "http://localhost:7700", apiKey: "masterKey", session: session)
-    let keyExpectation = XCTestExpectation(description: "Get all keys")
 
-    self.client.getKeys { result in
+    // remove all keys to keep a clean state
+    self.client.getKeys(params: KeysQuery(limit: 100, offset: 0)) { result in
       switch result {
       case .success(let keys):
-        if keys.results.count > 0 {
-          let key = keys.results.first
-          if let firstKey: Key = key {
-            self.key = firstKey.key
+        keys.results.forEach {
+          self.client.deleteKey(key: $0.uid) { result in
+            switch result {
+            case .success:
+              ()
+            case .failure(let error):
+              dump(error)
+              XCTFail("Failed to delete key")
+            }
           }
-        } else {
-          XCTFail("Failed to get keys")
         }
-        keyExpectation.fulfill()
       case .failure(let error):
         dump(error)
-        XCTFail("Failed to get keys")
-        keyExpectation.fulfill()
+        XCTFail("Failed to retrieve and delete all keys")
       }
     }
-    self.wait(for: [keyExpectation], timeout: TESTS_TIME_OUT)
   }
 
   func testGetKeys() {
-    let keyExpectation = XCTestExpectation(description: "Get all keys")
+    let keyExpectation = XCTestExpectation(description: "Get a list of keys")
 
-    self.client.getKeys { result in
+    self.client.createKey(KeyParams(actions: ["*"], indexes: ["*"], expiresAt: nil)) { result in
       switch result {
-      case .success(let keys):
-        XCTAssertNotEqual(keys.results.count, 0)
+      case .success:
+        self.client.getKeys { result in
+          switch result {
+          case .success(let keys):
+            XCTAssertNotEqual(keys.results.count, 0)
+          case .failure(let error):
+            dump(error)
+            XCTFail("Failed to get all keys")
+          }
+        }
+
         keyExpectation.fulfill()
       case .failure(let error):
         dump(error)
-        XCTFail("Failed to get all keys")
+        XCTFail("Failed to create a key")
         keyExpectation.fulfill()
       }
     }
+
     self.wait(for: [keyExpectation], timeout: TESTS_TIME_OUT)
   }
 
   func testGetKey() {
     let keyExpectation = XCTestExpectation(description: "Get one key")
 
-    self.client.getKey(key: self.key) { result in
+    self.client.createKey(KeyParams(actions: ["*"], indexes: ["*"], expiresAt: nil)) { result in
       switch result {
-      case .success(let key):
-        XCTAssertNotNil(key.description)
+      case .success(let createdKey):
+        self.client.getKey(key: createdKey.uid) { result in
```

Comment:
maybe `key` should be renamed `keyOrUid` ? As it seems we are not suggesting `uid` as a valid option with this parameter name